### DISCUSSION TITLE
Frequency control in AC power flow calculation?

### DISCUSSION THREAD

**Discussion by noadr98** [e2nIEE/pandapower#1762] [Q&A] [answered]
Dear all,

I was wondering, whether there is any built-in method to include frequency control in the standard AC power flow in PandaPower.
By default, the slack node obviously balances active and reactive power in the grid, however I would like to include another generator to contribute to the active power balance in the grid within its active power limits.
I couldn't find any information on that in the docs, and it also seems that this is not foreseen in any of the pre-defined controllers...

I would apprecaite any help on this question, thank you :).

**Comment by rbolgaryn** [comment, marked as answer]
Dear @noadr98 you should look into the distributed slack power flow. You can find a jupyter notebook in the tutorials

**Reply by noadr98** [reply]
Thanks for the answer. I wonder if there is any way to enforce generator active power limits in the distributed slack power flow?
